# Week 2 Checkpoint: First Data Profile
**BAN 6003 Data Management and Analytics Integration**

Complete this notebook using the `employee_data.csv` file in the `data/` folder.

This is the independent checkpoint for Week 2. Use the functions practiced in the guided lab. You are not expected to clean the data yet. Your goal is to profile the data and report what you notice.


## Instructions

1. Run each cell from top to bottom.
2. Fill in all cells marked `# Your code here`.
3. Write short answers in the markdown response areas.
4. Save, commit, and push your notebook to GitHub.

You may use the Week 2 guided lab as a reference.


## Task 0: Import Pandas and Load the Data
Run the cell below.


In [2]:
import pandas as pd

df = pd.read_csv("../data/employee_data.csv")
df.head()


,employee_id,first_name,last_name,hire_date,department,salary,age,full_time
0,101,John,Smith,4/15/22,Marketing,85000,34.0,True
1,102,Jane,Doe,8/20/21,Sales,92000,45.0,True
2,103,Peter,Jones,1/2/22,HR,71000,29.0,True
3,104,Mary,Johnson,11/10/23,Engineering,115000,38.0,True
4,105,Mike,Brown,5/30/20,sales,89000,NaN,False


## Task 1: Basic Structure
Show the basic structure of the DataFrame.

Your code should show:
- the shape of the DataFrame;
- the column names;
- the first five rows;
- the last five rows.


In [3]:
df.shape

(13, 8)

In [4]:
df.columns

Index(['employee_id', 'first_name', 'last_name', 'hire_date', 'department',
       'salary', 'age', 'full_time'],
      dtype='str')

In [5]:
df.head()

,employee_id,first_name,last_name,hire_date,department,salary,age,full_time
0,101,John,Smith,4/15/22,Marketing,85000,34.0,True
1,102,Jane,Doe,8/20/21,Sales,92000,45.0,True
2,103,Peter,Jones,1/2/22,HR,71000,29.0,True
3,104,Mary,Johnson,11/10/23,Engineering,115000,38.0,True
4,105,Mike,Brown,5/30/20,sales,89000,NaN,False


In [6]:
df.tail()

,employee_id,first_name,last_name,hire_date,department,salary,age,full_time
8,102,Emily,White,1/15/24,Sales,91000,31.0,True
9,110,David,Miller,7/22/19,HR,68000,58.0,True
10,111,Sarah,Moore,3/12/23,Marketing,"$91,000",28.0,True
11,error,Kevin,Clark,2/5/24,Sales,88000,35.0,True
12,113,Brian,Hall,6/18/22,Engineering,5900000,999.0,True


### Task 1 Written Response
Write 1–2 sentences describing the size and general structure of the dataset.

**Your response:**  
The dataset has 13 entries for each employee, as well as covers 6 items including ID, first and last name, hire date, department, salary, age, and if they are full time.


## Task 2: Data Types and Non-Missing Counts
Use Pandas to inspect data types and non-missing counts.

Your code should use:
- `info()`
- `dtypes`


In [7]:
df.info

<bound method DataFrame.info of    employee_id first_name last_name    hire_date   department    salary  \
0          101       John     Smith      4/15/22    Marketing     85000   
1          102       Jane       Doe      8/20/21        Sales     92000   
2          103      Peter     Jones       1/2/22           HR     71000   
3          104       Mary   Johnson     11/10/23  Engineering    115000   
4          105       Mike     Brown      5/30/20        sales     89000   
5          106      Susan     Davis          NaN        Sales     95000   
6          107      Chris    Wilson  2023-FEB-21    Marketing       NaN   
7          108      Laura    Taylor       9/1/21  Engineering    120000   
8          102      Emily     White      1/15/24        Sales     91000   
9          110      David    Miller      7/22/19           HR     68000   
10         111      Sarah     Moore      3/12/23    Marketing  $91,000    
11       error      Kevin     Clark       2/5/24        Sales     88

In [8]:
df.dtypes

employee_id        str
first_name         str
last_name          str
hire_date          str
department         str
salary             str
age            float64
full_time       object
dtype: object

In [9]:
# Your code here
df.isnull()

,employee_id,first_name,last_name,hire_date,department,salary,age,full_time
0,False,False,False,False,False,False,False,False
1,False,False,False,False,False,False,False,False
2,False,False,False,False,False,False,False,False
3,False,False,False,False,False,False,False,False
4,False,False,False,False,False,False,True,False
5,False,False,False,True,False,False,False,False
6,False,False,False,False,False,True,False,False
7,False,False,False,False,False,False,False,True
8,False,False,False,False,False,False,False,False
9,False,False,False,False,False,False,False,False


### Task 2 Written Response
Which columns, if any, have data types that might need to be investigated later? For example, is a number or date stored as text/object?

**Your response:**  
Salary and Age. Salary is not numeric and age is listing as a null

## Task 3: Missing Values
Check missing values for all columns.

Your code should use `isna().sum()`.

Optional: also calculate missing percentages.


In [10]:
# Your code here
df.isna().sum()

employee_id    0
first_name     0
last_name      0
hire_date      1
department     0
salary         1
age            1
full_time      1
dtype: int64

### Task 3 Written Response
Which columns have missing values? Which missing values seem most important to investigate later?

**Your response:**  
Hire date, salary, age, and full time

## Task 4: Numerical Summary
Use `.describe()` to summarize numerical columns.

Then choose one numerical column and calculate at least two individual summary statistics such as mean, median, min, max, or standard deviation.


In [18]:
# Your code here
df.describe()

,age
count,12.000000
mean,118.583333
std,277.411889
min,28.000000
25%,32.500000
50%,36.500000
75%,46.750000
max,999.000000


### Task 4 Written Response
What do you notice from the numerical summaries? Mention at least one range, unusual value, or pattern.

**Your response:**  
Age has a mean of 188.58 which is very odd. This is more than likely due to the outlier of the max age showing as 999. As well as since the age is the only column that has solely numerics it is the only result that is produced.

## Task 5: Categorical Summary
Use `.value_counts()` to summarize at least one categorical column.

Start with `department` if it exists in the dataset. Then choose one additional categorical column if available.


In [12]:
# Your code here
df.value_counts('department')

department
Sales          4
Marketing      3
Engineering    3
HR             2
sales          1
Name: count, dtype: int64

In [15]:
df.value_counts('full_time')

full_time
True     11
False     1
Name: count, dtype: int64

In [19]:
df.value_counts()

employee_id  first_name  last_name  hire_date  department   salary    age    full_time
101          John        Smith      4/15/22    Marketing    85000     34.0   True         1
102          Jane        Doe        8/20/21    Sales        92000     45.0   True         1
103          Peter       Jones      1/2/22     HR           71000     29.0   True         1
104          Mary        Johnson    11/10/23   Engineering  115000    38.0   True         1
102          Emily       White      1/15/24    Sales        91000     31.0   True         1
110          David       Miller     7/22/19    HR           68000     58.0   True         1
111          Sarah       Moore      3/12/23    Marketing    $91,000   28.0   True         1
error        Kevin       Clark      2/5/24     Sales        88000     35.0   True         1
113          Brian       Hall       6/18/22    Engineering  5900000   999.0  True         1
Name: count, dtype: int64

### Task 5 Written Response
What are the most common categories? Do you notice anything that might need standardization or further investigation later?

**Your response:**  
The most common category is full time employees according to the dataset.

## Task 6: Duplicate Checks
Check for duplicate records and duplicate employee IDs.

Your code should check:
- fully duplicated rows;
- duplicate `employee_id` values;
- the most common `employee_id` values using `value_counts()`.


In [20]:
# Your code here
df.value_counts('employee_id')

employee_id
102      2
101      1
103      1
104      1
105      1
106      1
107      1
108      1
110      1
111      1
error    1
113      1
Name: count, dtype: int64

### Task 6 Written Response
Are there duplicate employee IDs? If yes, what might that mean if `employee_id` is supposed to uniquely identify each employee?

**Your response:**  
There are two 102 employee ID's. That would either mean an error or it could mean that employee was rehired and retained their ID. Either way this will cause issues if you search solely using the ID.

## Task 7: Overall Data Quality Observations
Write 3–5 bullet points summarizing the condition of the dataset.

Focus on what you found from profiling. Do **not** clean the data yet.

Examples:
- The dataset has ___ rows and ___ columns.
- The ___ column has missing values.
- The ___ column may have an incorrect data type.
- The ___ column has possible duplicate IDs.
- The ___ column has a range from ___ to ___.


In [21]:
df.shape

(13, 8)

**Your observations:**

1. The data set has 13 rows and 8 columns
2. There are 4 null values in the data set
3. Column 5 Row 10 has incorrect data type for salary
4. [Optional Observation 4]
5. [Optional Observation 5]


## Submission Checklist
Before submitting, make sure:

- [x] All code cells run without errors.
- [x] You completed all tasks.
- [x] You wrote short responses in the markdown cells.
- [x] You saved the notebook.
- [x] You committed and pushed your work to GitHub.

Use the terminal:

```bash
git add .
git commit -m "Completed Week 2 data profile checkpoint"
git push
```
